File: ruby/day_of_the_week.rb
# Ruby 2.6.3
# https://leetcode.com/problems/day-of-the-week/

require 'date'

def day_of_the_week(day, month, year)
  # %A will return the full name of the day
  return Date.new(year, month, day).strftime("%A")
end

puts day_of_the_week(3, 7, 1994)

File: go/day_of_the_week.go
// https://leetcode.com/problems/day-of-the-week/

package main

import (
	"fmt"
	"time"
)

func dayOfTheWeek(day int, month int, year int) string {

	date := time.Date(year, time.Month(month), day, 0, 0, 0, 0, time.UTC)
	result := date.Weekday().String()

	return result
}

func main() {
	fmt.Println(dayOfTheWeek(31, 10, 2019))
}
